lib/version-check.js

```diff
@@ -34,18 +48,17 @@ module.exports = function (gulp, gutil) {
         return;
       }
       try {
-        var versionResponse = parseResponse(body, opts, errorCallback);
+        versionResponse = parseResponse(body, opts, errorCallback);
         var version = versionResponse.version;
         var matches = version.toString().match(versionPattern);
         if (!matches || matches.length < 3) {
           process.exitCode = 1;
-          callbackOrThrowError(errorCallback, 'Invalid version check response ' + version);
-          return;
+          return callbackOrThrowError(errorCallback, 'Invalid version check response ' + version);
         }
         var versionOnServer = parseInt(matches[1]);
         var minorVersionOnServer = parseInt(matches[2]);
         if (versionOnServer < serverVersion.supportedVersionMajor ||
-          (versionOnServer === serverVersion.supportedVersionMajor &&
+          (versionOnServer == serverVersion.supportedVersionMajor &&
```

Comment:
`===` check is safer than `==`

Reply:
yes, but `===` is comparing the data type as well, which ended up giving false here even when the above 2 values were equal, so had to change it to `==`

Reply:
You could try to convert one of them so that both of them have same type and then use '==='

Reply:
okay, will change accordingly to use `===`